blivet/devicetree.py

```diff
@@ -152,6 +152,13 @@ def _add_device(self, newdev, new=True):
             if parent not in self._devices:
                 raise DeviceTreeError("parent device not in tree")
 
+        if isinstance(newdev, LVMLogicalVolumeDevice) and newdev.from_lvs:
+            # if adding an LV constructed from other LVs, we may need to remove
+            # the LVs it's supposed to be constructed from from the device tree
+            for lv in newdev.from_lvs:
+                if lv in self._devices:
+                    self._remove_device(lv)
+
```

Comment:
If the code to do the reverse moves to `cancel_action` we'll obviously want to move this code into `_register_action`.